cmd/minishift/cmd/root.go

```diff
@@ -73,6 +75,13 @@ var RootCmd = &cobra.Command{
 			isAddonInstallRequired bool
 		)
 
+		// If profileName is not an empty string then we need to update
+		// constants.ProfileName to make sure it will create requested directory structure.
+		profileName := isProfileExist(cmd)
```

Comment:
Much better! Reads quite well. Thanks.

Reply:
>I do not think we need to return profileName from isProfileExist(cmd). Because at this stage constant.ProfileName would have the correct profile name. The only thing we need to do is, we need to check if --profile PROFILE_NAME is a valid profile (with directory structure).

@LalatenduMohanty No at this point `constant.ProfileName` only have either active profile info or default profile. It doesn't have any info about `--profile` flag parameter until we set it.

> I understand that we need to do it in PersistentPreRun because we want to take help of Viper to see if it is part of minishift start --profile.
In initializeProfile(), the --profile takes priority than active profile. So it will always return the correct profile name. If it returns null then constants.ProfileName would have default profile name.

@LalatenduMohanty Sorry, here you already contradicting your first argument. You are absolutely right but to get it we return the parameter which is part of profile flag.

> isProfileExist sounds like a function which returns boolean value. Also the name does not communicate the intent of the function.

Agree, might be `setExistProfile` => return profile name if exist?